### PR TITLE
Optimize `winVersion.isWin10` (#10335)

### LINKED ISSUE #10335: winVersion.isWin10: modify it to return early if major is not 10
Hi,

This feature request should not be done for 2019.3 in order to avoid introducing possible regressions, but perhaps defer it to 2020.1:

### Is your feature request related to a problem? Please describe.
In NVDA 2019.2, as part of UIA console support, winVersion module introduced isWin10 function which returns whether a system is running a specific version of Windows 10 or later. At the moment it checks major version in two places:

* If "at least" flag is specified: return False if version is less than 10.
* If "at least" flag is not specified: return False if major version is not 10.

At least it is understandable. But I would argue that for sake of performance and keeping the subject line and body in sync, I propose the below modification.

### Describe the solution you'd like
Instead of checking flags after defining version information map, return early if major version isn't even 10. Not only flag checks can be avoided on Windows 8.1 and earlier, it can make the function body in line with what the function title says.

Thus I propose:

1. First, check winVersion.major and return False if it isn't 10.
2. Define version to build map.
3. Same as current function.

### Describe alternatives you've considered
Keep the function as is.

### Additional context
I'm sure there might be other opportunities to optimize this function, but that's in the future.

Thanks.

### PR DESCRIPTION
**Pull request by JulienCochuyt**
### Link to issue number:

Fixes #10335

### Summary of the issue:

As pointed out by @josephsl in https://github.com/nvaccess/nvda/issues/10335#issue-503035051, `winVersion.isWin10` can be improved to at least return early in Windows 7 and 8.1

### Description of how this pull request fixes the issue:

As proposed by @josephsl, return early with `False` if major version is not 10.
Additionally, for further slight performance improvement, factor out the mapping of versions to build number and import late from `logHandler` as needed.

### Testing performed:

Flagging this PR as draft for now for thorough testing with a proper try-build.

### Known issues with pull request:

The outcome of the function would only differ with a greater Windows major version, which is not planned in a near future.
An alternative would then be to first test for `winVersion.major < 10` instead of `winVersion.major != 10`, but versions to build mappings would need to be checked then anyway.

### Change log entry:

I don't think this deserves to be announced.

### CODE REVIEW

**Comment by josephsl** [comment]
CC @CodeOfDusk

**Comment by leonardder** [comment]
@JulienCochuyt Do you want an official try build, or is the automatic pr build sufficient?

File: source/winVersion.py [hunk 1]
```diff
@@ -27,33 +27,34 @@ def canRunVc2010Builds():
...
+
+WIN10_VERSIONS_TO_BUILDS = {
+	1507: 10240,
+	1511: 10586,
+	1607: 14393,
+	1703: 15063,
+	1709: 16299,
+	1803: 17134,
+	1809: 17763,
+	1903: 18362,
```
Comment (leonardder) [inline source/winVersion.py]:
```suggestion
	1903: 18362,
	1909: 18363,
```

**Comment by JulienCochuyt** [comment]
> @JulienCochuyt Do you want an official try build, or is the automatic pr build sufficient?

I guess the automatic one is sufficient, but I'm not the most knowledgeable on this topic.
@josephsl, @codeofdusk do you agree or do you have reasons to require a full blown official try build here?

**Comment by josephsl** [comment]
Hi, I think a PR build is sufficient for now until you say it is ready for an official review (unless the draft status has changed). Also, thanks for adding 1909 – I meant to submit a new issue, but at least we’ve taken care of it as part of this, so works out well. Thanks.

**Review by feerrenrut — APPROVED** [review]
Thanks @JulienCochuyt

**Pull request merged**